stl/inc/ranges

```diff
@@ -8399,8 +8399,14 @@ namespace ranges {
         }
     }
 
+    // clang-format off
+    template <class _Container>
+    concept _Is_cv_unqualified = !is_const_v<_Container> && !is_volatile_v<_Container>;
+    // clang-format on
+
+
     _EXPORT_STD template <class _Container, input_range _Rng, class... _Types>
-        requires (!view<_Container>)
+        requires (_Is_cv_unqualified<_Container> && !view<_Container>)
     _NODISCARD constexpr _Container to(_Rng&& _Range, _Types&&... _Args) {
         if constexpr (_Converts_direct_constructible<_Rng, _Container, _Types...>) {

```

Comment:
I think `static_assert` is preferred, although additional constraints are also valid as long as they don't make another overload selected. (Same for the other overload.)

```suggestion
        requires (!view<_Container>)
    _NODISCARD constexpr _Container to(_Rng&& _Range, _Types&&... _Args) {
        static_assert(!is_const_v<_Container> && !is_volatile_v<_Container>, "the target type must be cv-unqualified");
        if constexpr (_Converts_direct_constructible<_Rng, _Container, _Types...>) {
```

Reply:
As far as I remember we were advised to avoid `&&` in `static_assert`s and `assert`s
So maybe two `static_assert` will be better.

Reply:
In this case, out of curiosity, why not just put the `!view<_Container>` in an assert as well? 

I'm assuming it's because the constraint changes what operator ends up chosen (Which we don't want to change) and the static assert doesn't?

Reply:
I guess that also gets into a bigger question, why would we rather have something hit a static assert instead of potentially overloading to another function that might be viable?

Reply:
>why would we rather have something hit a static assert instead of potentially overloading to another function that might be viable?

We can write a good error message for `static_assert`.

Reply:
Isn't that a large motivation behind concepts?

Reply:
Yes 😺 , but MSVC has bad messages. We have an issue about it: https://github.com/microsoft/STL/issues/2648

Reply:
> As far as I remember we were advised to avoid `&&` in `static_assert`s and `assert`s

I don't think there's a general reason to avoid conjunctions in `static_assert`s. We wouldn't want to make a user guess (`static_assert(is_meow<T> && is_meow<U>, "T and U must both be meow")`) when we could be more specific with two `static_assert`s, but something like `static_assert(!is_const_v<T> && !is_volatile_v<T>, "T must be a cv-unqualified type")` would be fine.